pkg/virt-controller/services/net_binding_test.go

```diff
@@ -0,0 +1,85 @@
+package services_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/hooks"
+	"kubevirt.io/kubevirt/pkg/virt-controller/services"
+	"kubevirt.io/kubevirt/tests/libvmi"
+)
+
+var _ = Describe("Network Binding", func() {
+	const (
+		testNetworkName1  = "net1"
+		testBindingName1  = "binding1"
+		testSidecarImage1 = "image1"
+		testNetworkName2  = "net2"
+		testBindingName2  = "binding2"
+		testSidecarImage2 = "image2"
+		testNetworkName3  = "net1"
+
+		shouldSucceed = true
+	)
+
+	BeforeEach(func() {
+
+	})
+	Context("binding plugin sidecar list", func() {
+		DescribeTable("should create the correct sidecars", func(vmi *v1.VirtualMachineInstance, bindings map[string]v1.InterfaceBindingPlugin, expectedSidecars hooks.HookSidecarList, shouldSucceed bool) {
+			config := &v1.KubeVirtConfiguration{
+				NetworkConfiguration: &v1.NetworkConfiguration{
+					Binding: bindings,
+				},
+			}
+			sidecars, err := services.BindingPluginSidecarList(vmi, config)
+			if shouldSucceed {
+				Expect(err).ToNot(HaveOccurred())
+				Expect(sidecars).To(Equal(expectedSidecars))
+			} else {
+				Expect(err).To(HaveOccurred())
+			}
+		},
+			Entry("VMI has binding plugin but config doesn't exist",
+				libvmi.New(libvmi.WithInterface(v1.Interface{Name: testNetworkName1, Binding: &v1.PluggedBinding{Name: testBindingName1}}),
```

Comment:
I think it was raised before, we should not use e2e tests helpers in unit tests, isn't it? (talking about libvmi)